### PR TITLE
parser: handle PARTITION keyword generated by mydumper

### PR DESCRIPTION
**Pull request by tiancaiamao**
naive support for mydumper generated file, PARTITION is parsed and ignore. 

For table create statement like:

```
CREATE TABLE `company_modify` (`id` int(11) NOT NULL AUTO_INCREMENT,`entid` int(11) NOT NULL DEFAULT '0',`ALTITEM` varchar(3) DEFAULT NULL COMMENT '变更事项',`ALTBE` text COMMENT '变更前内容',`ALTAF` text COMMENT '变更后内容',`ALTDATE` date DEFAULT NULL COMMENT '变更日期',`ALTTIME` int(11) DEFAULT NULL COMMENT '变更次数',`updated` datetime DEFAULT NULL,PRIMARY KEY (`id`,`entid`),KEY `entid` (`entid`),KEY `ALTITEM` (`ALTITEM`)) ENGINE=MyISAM AUTO_INCREMENT=96413735 DEFAULT CHARSET=utf8 ROW_FORMAT=DYNAMIC COMMENT='变 更记录'
/*!50100 PARTITION BY HASH (entid)PARTITIONS 32 */;
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
@qiuyesuifeng

File: parser/parser_test.go [hunk 1]
```diff
@@ -864,6 +864,8 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"drop table if exists xxx", true},
 		{"drop table if not exists xxx", false},
 		{"drop view if exists xxx", true},
+		// Partition option generated by mydumper
+		{"CREATE TABLE `company_modify` (`id` int(11) NOT NULL AUTO_INCREMENT,`entid` int(11) NOT NULL DEFAULT '0',`ALTITEM` varchar(3) DEFAULT NULL COMMENT '变更事项',`ALTBE` text COMMENT '变更前内容',`ALTAF` text COMMENT '变更后内容',`ALTDATE` date DEFAULT NULL COMMENT '变更日期',`ALTTIME` int(11) DEFAULT NULL COMMENT '变更次数',`updated` datetime DEFAULT NULL,PRIMARY KEY (`id`,`entid`),KEY `entid` (`entid`),KEY `ALTITEM` (`ALTITEM`)) ENGINE=MyISAM AUTO_INCREMENT=96413735 DEFAULT CHARSET=utf8 ROW_FORMAT=DYNAMIC COMMENT='变 更记录'/*!50100 PARTITION BY HASH (entid)PARTITIONS 32 */;", true},
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: I prefer to use a common test case.

**Comment by shenli** [comment]
LGTM

File: parser/parser_test.go [hunk 2]
```diff
@@ -864,6 +864,8 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"drop table if exists xxx", true},
 		{"drop table if not exists xxx", false},
 		{"drop view if exists xxx", true},
+		// Partition option generated by mydumper
```
Comment (ngaut) [inline parser/parser_test.go]: Please simplify the test.

**Comment by coocood** [comment]
LGTM

**Pull request merged**